src/cunumeric/matrix/contract.cc

```diff
@@ -33,23 +33,37 @@ using namespace tblis;
 // code. These types are bit-identical, so we can safely cast from one to the other in host code,
 // to appease the type checker.
 
+namespace {  // anonymous
+
+// Contract type helper ensures that the types of shapes, strides and modes we pass through
+// to TBLIS calls match up with the types that TBLIS expects. Some (strict) compilers will
+// throw an error trying to implicitly cast between tblis::len_type and int64_t even though
+// they are the same size.
+struct ContractTypeHelper {
+  typedef tblis::len_type len_type;
```

Comment:
a nitpick: let's use `using` instead

Reply:
also, what I suggested was using different `len_type`s for different variants so no castings would be necessary. For the gpu variant, you can use the following and no castings will be needed:
```
struct GPUContractTypeHelper {
  using len_type = int64_t;
  using stride_type = int64_t;
  using label_type = int32_t;
};
```
you could also inline these members in `ContractImplBody`